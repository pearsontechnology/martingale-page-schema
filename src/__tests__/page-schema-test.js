// page-schema-test.js
import React from 'react';
import {pageSchemaToReact} from '../';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';

const Greet=({name='World'})=>(<div>Hello {name}!</div>);
Greet.propTypes={
  name: PropTypes.string
};
const GreetList=({people=[]})=><div>{people.map((person, index)=><Greet key={index} {...person} />)}</div>;

const data = {
  name: 'Test',
  people: [
    {name: 'Bob'},
    {name: 'Sue'},
    {name: 'Phil'},
    {name: 'Henry'},
  ]
};

const layout={
  type: 'div',
  children: [
    {
      type: 'Greet'
    },
    {
      type: 'Greet',
      props: {
        name: 'Static'
      }
    },
    {
      type: 'Greet',
      props: {
        name: data.name
      }
    },
    {
      type: 'Greet',
      props: data
    },
    {
      type: 'GreetList',
      props: data
    }
  ]
};

const components={
  Greet,
  GreetList
};

test('pageSchemaToReact', () => {
  const component = renderer.create(
    pageSchemaToReact({layout, components})
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
