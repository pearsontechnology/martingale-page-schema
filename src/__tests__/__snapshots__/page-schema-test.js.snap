// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pageSchemaToReact 1`] = `
<div>
  <div>
    Hello 
    World
    !
  </div>
  <div>
    Hello 
    Static
    !
  </div>
  <div>
    Hello 
    Test
    !
  </div>
  <div>
    Hello 
    Test
    !
  </div>
  <div>
    <div>
      Hello 
      Bob
      !
    </div>
    <div>
      Hello 
      Sue
      !
    </div>
    <div>
      Hello 
      Phil
      !
    </div>
    <div>
      Hello 
      Henry
      !
    </div>
  </div>
</div>
`;
